src/cloudflare/internal/vectorize-api.ts

```diff
@@ -0,0 +1,203 @@
+          accept: "application/json",
+        },
+      }
+    );
+
+    return await toJson<VectorizeMatches>(res);
+  }
+
+  public async insert(
+    vectors: VectorizeVector[]
+  ): Promise<VectorizeVectorMutation> {
+    const res = await this._send(
+      Operation.VECTOR_INSERT,
+      `indexes/${this.indexId}/insert`,
+      {
+        method: "POST",
+        body: JSON.stringify({
+          vectors: vectors.map((vec) => ({
+            ...vec,
+            values: Array.isArray(vec.values)
+              ? vec.values
+              : Array.from(vec.values),
+          })),
+        }),
+        headers: {
+          "content-type": "application/json",
+          "cf-vector-search-dim-width": String(
+            vectors.length ? vectors[0]?.values?.length : 0
+          ),
+          "cf-vector-search-dim-height": String(vectors.length),
+          accept: "application/json",
+        },
+      }
+    );
+
+    return await toJson<VectorizeVectorMutation>(res);
+  }
+
+  public async upsert(
+    vectors: VectorizeVector[]
+  ): Promise<VectorizeVectorMutation> {
+    const res = await this._send(
+      Operation.VECTOR_UPSERT,
+      `indexes/${this.indexId}/upsert`,
+      {
+        method: "POST",
+        body: JSON.stringify({
+          vectors: vectors.map((vec) => ({
+            ...vec,
+            values: Array.isArray(vec.values)
+              ? vec.values
+              : Array.from(vec.values),
+          })),
+        }),
+        headers: {
+          "content-type": "application/json",
+          "cf-vector-search-dim-width": String(
+            vectors.length ? vectors[0]?.values?.length : 0
+          ),
+          "cf-vector-search-dim-height": String(vectors.length),
+          accept: "application/json",
+        },
+      }
+    );
+
+    return await toJson<VectorizeVectorMutation>(res);
+  }
+
+  public async getByIds(ids: string[]): Promise<VectorizeVector[]> {
+    const res = await this._send(
+      Operation.VECTOR_GET,
+      `indexes/${this.indexId}/getByIds`,
+      {
+        method: "POST",
+        body: JSON.stringify(ids),
+        headers: {
+          "content-type": "application/json",
+          accept: "application/json",
+        },
+      }
+    );
+
+    return await toJson<VectorizeVector[]>(res);
+  }
+
+  public async deleteByIds(ids: string[]): Promise<VectorizeVectorMutation> {
+    const res = await this._send(
+      Operation.VECTOR_DELETE,
+      `indexes/${this.indexId}/deleteByIds`,
+      {
+        method: "POST",
+        body: JSON.stringify(ids),
+        headers: {
+          "content-type": "application/json",
+          accept: "application/json",
+        },
+      }
+    );
+
+    return await toJson<VectorizeVectorMutation>(res);
+  }
+
+  private async _send(
+    operation: Operation,
+    endpoint: string,
+    init: RequestInit
+  ): Promise<Response> {
+    const res = await this.fetcher.fetch(
+      `http://vector-search/binding/${endpoint}`,
```

Comment:
I had commented on this but I think there was an update while I adding the comment :-) ... it would be helpful to add a comment here that `http://vector-search` here is a non-functional placeholder rather than a real url to a real site.